stdlib/public/core/StringBridge.swift

```diff
@@ -423,25 +423,37 @@ private var expectedConstantTagValue:UInt {
 }
 #endif
 
+private enum ConstantTaggedStringValidationResult {
+  case notApplicable
+  case valid
+  case invalid
+}
+
 @inline(__always)
-private func formConstantTaggedCocoaString(
-  untaggedCocoa: _CocoaString
-) -> AnyObject? {
-#if !arch(arm64)
-  return nil
-#else
+private func validateConstantTaggedCocoa(_ cocoaString: _CocoaString) ->
+  ConstantTaggedStringValidationResult {
+  #if !arch(arm64)
+  return .notApplicable
+  #else
+  guard _isObjCTaggedPointer(cocoaString) else {
+    return .notApplicable
+  }
 
-  let constantPtr:UnsafeRawPointer = Builtin.reinterpretCast(untaggedCocoa)
+  let taggedValue = unsafeBitCast(cocoaString, to: UInt.self)
 
-  // Check if what we're pointing to is actually a valid tagged constant
-  guard _swift_stdlib_dyld_is_objc_constant_string(constantPtr) == 1 else {
-    return nil
+  guard taggedValue & constantTagMask == expectedConstantTagValue else {
+    return .notApplicable
   }
 
-  let retaggedPointer = UInt(bitPattern: constantPtr) | expectedConstantTagValue
+  let payloadMask = ~constantTagMask
+  let payload = taggedValue & payloadMask
```

Comment:
Is it always the case that all other bits are part of the address, or just the bottom 60?